toolchain/sem_ir/typed_insts.h

```diff
@@ -302,6 +302,17 @@ struct BindValue {
   InstId value_id;
 };
 
+// Represents a binding pattern value with name `entity_name_id`.
+// `bind_inst_id` is the instruction which actually binds the value.
+struct BindingPattern {
+  static constexpr auto Kind = InstKind::BindingPattern.Define<Parse::NodeId>(
+      {.ir_name = "binding_pattern"});
+
+  TypeId type_id;
+  EntityNameId entity_name_id;
```

Comment:
Sorry, one more question. It looks like each place this is set, it's actually coming from `bind_inst_id`. Why is it provided in both places?